src/infer_request.cc

```diff
@@ -682,6 +714,59 @@ InferenceRequest::Normalize()
 {
   const inference::ModelConfig& model_config = model_raw_->Config();
 
+  // Fill metadata for raw input
+  if (!raw_input_name_.empty()) {
+    if (original_inputs_.size() != 1) {
+      return Status(
+          Status::Code::INVALID_ARG,
+          "Raw request must only have 1 input to be deduced but got " +
+              std::to_string(original_inputs_.size()) + " inputs for model '" +
+              ModelName() + "'");
+    }
+    auto it = original_inputs_.begin();
+    if (raw_input_name_ != it->first) {
+      return Status(
+          Status::Code::INVALID_ARG,
+          "Unexpected reference name for raw input '" + raw_input_name_ +
+              "' got '" + it->first + "'");
+    }
+    const auto& config_input = model_config.input(0);
+    auto& raw_input = it->second;
+    std::vector<int64_t> shape;
+    if (model_config.max_batch_size() != 0) {
+      shape.emplace_back(1);
+    }
+    int64_t dynamic_axis = -1;
+    size_t element_cnt = 1;
+    for (const auto& dim : config_input.dims()) {
+      if (dim == WILDCARD_DIM) {
+        if (dynamic_axis != -1) {
+          return Status(
+              Status::Code::INVALID_ARG,
+              "The shape of the raw input '" + raw_input_name_ +
+                  "' can not be deduced because there are more than two "
+                  "dynamic dimension");
```

Comment:
I think we consistently use the term "variable-sized dimension" instead of "dynamic dimension"